scripts/boxgame.js

```diff
@@ -0,0 +1,119 @@
+class BoxGameComponent {
+    box = {};
+    score = document.createElement("div");
+    squares = [];
+    container = document.createElement('div');
+    gameAreaEl = document.createElement("div");
+    gamebox = {
+        x: Math.floor(800 / 100),
+        y: Math.floor(200 / 100)
+    };
+    player = {
+        speed: 100,
+        square: 1,
+        score: 0
+    };
+
+    init = (button) => {
+        this.gameAreaEl.innerHTML = '';
+        this.container.innerHTML = '';
+        this.score.innerHTML = '';
+        document.body.appendChild(this.container);
+        this.createGameAreaDom();
+        this.addListeners();
+        this.build();
+        button.style.display = 'none';
+    };
```

Comment:
i would use standard es6 class method sytax instead of field declarations
defining class properties like this is an experimental feature that needs transpiling with a babel plugin: https://medium.com/@charpeni/arrow-functions-in-class-properties-might-not-be-as-great-as-we-think-3b3551c440b1